eta/detectors/tfmodels_detectors.py

```diff
@@ -28,77 +28,81 @@
+    any omitted fields present in the default deployment config for the model
+    will be automatically populated.
+
+    Attributes:
+        model_name: the name of the published model to load
+        labels_path: the path to the labels map for the model
     '''
 
     def __init__(self, d):
         self.model_name = self.parse_string(d, "model_name")
+
+        # Loads any default deployment parameters
+        d = self.load_default_deployment_params(d, self.model_name)
+
         self.labels_path = etau.fill_config_patterns(
             self.parse_string(d, "labels_path"))
 
-    @classmethod
-    def load_default(cls):
-        '''Loads the default TFModelsDetectorConfig.'''
-        return cls.from_json(DEFAULT_CONFIG)
 
+class TFModelsDetector(ObjectDetector, UsesTFSession):
+    '''Interface to the TF-Models object detection library at
+    https://github.com/tensorflow/models/tree/master/research/object_detection.
 
-class TFModelsDetector(ObjectDetector):
-    '''Interface to the `tensorflow/models` object detection library.
+    This class uses `eta.core.tfutils.UsesTFSession` to create TF sessions, so
+    it automatically applies settings in your `eta.config.tf_config`.
 
-    https://github.com/tensorflow/models/tree/master/research/object_detection.
+    Instances of this class must either use the context manager interface or
+    manually call `close()` when finished to release memory.
     '''
 
-    def __init__(self, config=None):
-        '''Constructs a TFModelsDetector instance.
+    def __init__(self, config):
+        '''Creates a TFModelsDetector instance.
 
         Args:
-            config: an optional TFModelsDetectorConfig instance. If absent, the
-                default TFModelsDetectorConfig will be used
+            config: a TFModelsDetectorConfig instance
         '''
-        self.config = config or TFModelsDetectorConfig.load_default()
+        self.config = config
+        UsesTFSession.__init__(self)
 
         # Only downloads the model if necessary
         model_path = etam.download_model(self.config.model_name)
 
         # Load model
         self._tf_graph = self._build_graph(model_path)
-        self._sess = tf.Session(graph=self._tf_graph)
+        self._sess = self.make_tf_session(graph=self._tf_graph)
 
         # Load labels
         label_map = gool.load_labelmap(self.config.labels_path)
         categories = gool.convert_label_map_to_categories(
             label_map, max_num_classes=90, use_display_name=True)
         self._category_index = gool.create_category_index(categories)
 
-    @staticmethod
-    def _build_graph(model_path):
-        tf_graph = tf.Graph()
-        with tf_graph.as_default():
-            graph_def = tf.GraphDef()
-            with tf.gfile.GFile(model_path, "rb") as f:
-                graph_def.ParseFromString(f.read())
-                tf.import_graph_def(graph_def, name="")
-        return tf_graph
+    def __enter__(self):
```

Comment:
Why do we need these methods if it already inherits from `UsesTFSession`?

Reply:
True. I just did it for clarity to emphasize that it should be used as a context manager